LEGO1/lego/legoomni/src/input/legoinputmanager.cpp

```diff
@@ -153,9 +153,7 @@ MxResult LegoInputManager::GetJoystick()
 	}
 
 	MxS32 numJoysticks = 0;
-	if (m_joyids == NULL) {
-		m_joyids = SDL_GetJoysticks(&numJoysticks);
-	}
+	m_joyids = SDL_GetJoysticks(&numJoysticks);
```

Comment:
If you need to update the id each time then you have to free the old one first.
```suggestion
	if (m_joyids) {
		SDL_free(m_joyids);
	}
	m_joyids = SDL_GetJoysticks(&numJoysticks);
```